boards/posix/native_posix/doc/board.rst

```diff
@@ -538,11 +540,40 @@ The following peripherals are currently provided with this board:
   must be powered down and support Bluetooth Low Energy (i.e. support the
   Bluetooth specification version 4.0 or greater).
 
-Shell support
-*************
+**UART**
+  An optional UART driver can be compiled with native_posix.
+  For more information refer to the section `UART`_.
+
+
+UART
+*****
+
+This driver can be configured to either create and link the UART to a new
+pseudoterminal (i.e. ``/dev/pts<nbr>``), or to map the UART input and
+output to the executable's ``stdin`` and ``stdout``.
+This is chosen by selecting either
+:option:`CONFIG_NATIVE_UART_0_ON_OWN_PTY` or
+:option:`CONFIG_NATIVE_UART_0_ON_STDINOUT`
+Note that for interactive use, the first option should be chosen. The
+second option is only intended for automated testing or feeding/piping other
+processes output to the UART.
+
+When :option:`CONFIG_NATIVE_UART_0_ON_OWN_PTY` is chosen, the name of the
+newly created UART pseudo-terminal will be displayed in the console.
+You may also chose to automatically attach a terminal emulator to it by
+passing the command line option ``-attach_uart`` to the executable.
+The command used for attaching to the new shell can be set with the command line
+option ``-attach_uart_cmd=<"cmd">``. Where the default command is given by
+:option:`CONFIG_NATIVE_UART_AUTOATTACH_DEFAULT_CMD`.
+Note that the default command assumes both ``xterm`` and ``screen`` are
+installed in the system.
+
+
+Legacy shell support
+********************
 
-When the :ref:`Shell` subsystem is compiled with your application, the native
-standard input (``stdin``) will be redirected to the shell.
+When the legacy :ref:`Shell` subsystem is compiled with your application,
```

Comment:
``:ref:`shell` `` will be a reference to the **new** shell documentation and not to the **legacy** shell subsystem.  The shell documentation is changing with PR #10216, so your PR's references may break when 10216 is merged (or not work until it is). This should probably change to:

    When the :ref:`legacy shell` subsystem is compiled ...

Reply:
Yep.. I saw that PR coming after I sent mine. So I guess my options are 
a. to leave the old link which is correct today, and if that PR is merged before mine, correct mine
b. remove all links to ``Shell`` all together
c. Point to the new shell doc and stall this PR.. => don't like this one.